Mathlib/Algebra/Module/LinearMap/Pi.lean

```diff
@@ -0,0 +1,43 @@
+/-
+Copyright (c) 2019 Alexander Bentkamp. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Tomas Skrivan
+-/
+
+import Mathlib.Algebra.Module.Pi
+import Mathlib.Tactic.Abel
+import Mathlib.Algebra.Module.LinearMap.Defs
+import Mathlib.Algebra.Module.LinearMap.Prod
+
+/-!
+# Point evaluation is linear map
+
+Note that these results use `IsLinearMap`, which is mostly discouraged.
+
+## Tags
+linear algebra, vector space, module
+
+-/
+
+variable {R : Type*} {M N : Type*} {ι : Type*} [Semiring R]
+  [AddCommMonoid M] [Module R M]
+  [AddCommMonoid N] [Module R N]
+
+namespace IsLinearMap
+
+@[fun_prop]
+theorem isLinearMap_pi {f : M → ι → N}
+    (hf : ∀ i, IsLinearMap R (f · i)) :
+    IsLinearMap R (fun x i ↦ f x i) := by
+  apply IsLinearMap.mk
+  · intro x y; funext i
+    simp [(hf i).1]
+  · intro x y; funext i
+    simp [(hf i).2]
+
```

Comment:
This is obviously repeating the work of `LinearMap.pi`. Can we use one to prove the other?

Obviously the "cleaner" thing to do would be to redefine `LinearMap`, but that's a massive refactor we should not attempt unless we want to commit to doing it globally.

So in the meantime, I think you can use `(LinearMap.pi fun i => (hf i).mk _).isLinear` etc?

Reply:
Yes, I agree that the proof should be `(LinearMap.pi fun i => (hf i).mk _).isLinear` I did the proof the way I did because I'm not too  experienced using `LinearMap`

Reply:
`LinearMap.pi` lives in `Mathlib.LinearAlgebra.Pi`.

Similarly, to refactor proofs in `Mathlib.Algebra.Module.LinearMap.Prod` I would have to import `Mathlib.LinearAlgebra.Prod`. 

Is it ok to have imports from `Mathlib.Algebra`  in `Mathlib.LinearAlgebra` and vice versa?

Or should I move all those theorem to `Mathlib.LinearAlgebra`? But then, by Kevin's reaction on Zulip, it looks like that `IsLinearMap` is effectively deprecated so I'm not sure if it is desirable to expand the API.

Reply:
Given the premise of this PR is to expand the `IsLinearMap` API, I think it's best to do in the least duplicative way possible, which for now means adding the imports. We can move things around in review later.